litmus/director/openebs-self-install/test.yml

```diff
@@ -0,0 +1,187 @@
+---
+- hosts: localhost
+  connection: local
+
+  vars_files:
+    - test_vars.yml
+
+  tasks:
+  
+    ## Generating the testname for deployment
+    - include_tasks: /ansible-utils/create_testname.yml
+
+    ## Record Start-Of-Test In Litmus Result CR
+    - include_tasks: /ansible-utils/update_litmus_result_resource.yml
+      vars:
+        status: 'SOT'
+      
+    ## Saving the director url
+    - set_fact:
+        director_url : "http://{{ director_ip }}:30380"
+    
+    ## Getting username
+    - name: Get username
+      shell: cat /etc/secret-volume/username
+      register: username
+
+    ## Getting password     
+    - name: Get password
+      shell: cat /etc/secret-volume/password
+      register: password
+
+    ## Checking if openebs is already installed
+    - name: Check if openebs is already installed
+      shell: kubectl get deploy maya-apiserver --all-namespaces | wc -l
+      args:
+        executable: /bin/bash
+      register: maya_api
+      failed_when: "{{ maya_api.stdout }} != 0"
+
+    ## Fetch the projectid of the project where the cluster is running
+    - name: Fetch the project id of the project where the cluster is running
+      uri:
+        url: "{{ director_url }}/v3/groups/{{ group_id }}/project"
+        method: GET
+        url_username: "{{ username.stdout }}"
+        url_password: "{{ password.stdout }}"
+        force_basic_auth: yes
+        return_content: yes
+        status_code: 200
```

Comment:
why is status_code required ?

Reply:
So whatever response we get as a result of  `GET` or `POST` request it comes with status code which is basically predefined and we catch them using `status code`  field in ansible `uri` module. status code `200` is default status code if I remove this then also it will not make any difference.